### PR TITLE
Make firefox add-on build reproducible by fetching specific commit of code-intel-extensions

### PR DESCRIPTION
**Pull request by marekweb**
This is a set of improvements to the build process for the FF add-on with bundled extensions, that makes the whole process much easier.

We now fetch and build the `code-intel-extensions` repository as a zip at a specific commit.

A configuration file now specifies the revision of `code-intel-extensions`. We pin a specific revision so that the build is reproducible. This could be done with a git submodule as an alternative, but this method makes for a simpler build step.

```js
{
  // the revision of `sourcegraph/code-intel-extensions` to build from
  "revision": "282caa906a8f5d64f67711809a2c8492908f9fe4",
  // the list of extensions to bundle
  "extensions": ["template"]
}
```

In order to be able to zip the source code as a bundle that includes `sourcegraph/sourcegraph` along with `code-intel-extensions` (see #12918) the build step needs to be split into two steps: fetch and build.

**Roadmap**

- [x] implement `yarn run fetch-code-intel-extensions`
  - [x] Fetch `code-intel-extensions` source code for the given revision
- [x] implement `yarn run build-inline-extensions`
  - [x] Check that `code-intel-extensions` has been fetched
  - [x] Build from the fetched source code

**Blocking issues**

Blocking issues have been resolved. No longer blocked on anything.

### CODE REVIEW

**Comment by davejrt** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.20 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you

**Comment by marekweb** [comment]
This is now unblocked thanks to https://github.com/sourcegraph/code-intel-extensions/pull/492

File: browser/scripts/fetch-code-intel-extensions.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const shelljs = require('shelljs')
+const path = require('path')
+const signale = require('signale')
+const bundledCodeIntelExtensionsConfig = require('../bundled-code-intel-extensions.json')
+const temporarySourceDirectory = path.join(process.cwd(), 'code-intel-extensions22')
```
Comment (tjkandala) [inline browser/scripts/fetch-code-intel-extensions.js:5]: Is '22' intentional? If so, we should add it to 'build-inline-extensions. If not, we should remove it. The naming inconsistency is causing the build to fail ('Code intel extension "template" was not found for bundling.')

Comment (marekweb) [reply browser/scripts/fetch-code-intel-extensions.js]: That was a typo that slipped in on the last commit. Thanks, it's now fixed.

**Review by tjkandala — APPROVED** [review]

**Pull request merged**